CoreFoundation/RunLoop.subproj/CFSocket.c

```diff
@@ -213,13 +213,30 @@ CF_INLINE int __CFSocketLastError(void) {
 }
 
 CF_INLINE CFIndex __CFSocketFdGetSize(CFDataRef fdSet) {
+#if TARGET_OS_WIN32
+    if (CFDataGetLength(fdSet) == 0) {
+        return 0;
+    }
+    return FD_SETSIZE;
```

Comment:
My understanding of the structure on Windows is that it has a default size but it's legit to allocate a bigger buffer and treat the array as bigger than `FD_SETSIZE`. I think a better implementation would calculate the number of entries from `CFDataGetLength` after removing the size of the count field. This would also avoid the implicit dependency where this code is assuming that the data length is at least that of `fd_set`.

Reply:
Ahh, I think I will update the patch and exclude this function from Win build at all. It is not used in any of Windows code paths except one place where its result is ignored anyway.

Reply:
Resolved per convo about large FD_SETSIZE override